service_setup/matrix.md

```diff
@@ -0,0 +1,297 @@
+	    ssl_prefer_server_ciphers on;
+
+	    # HSTS (ngx_http_headers_module is required) (15768000 seconds = 6 months)
+	    add_header Strict-Transport-Security max-age=15768000;
+
+	    # OCSP Stapling: fetch OCSP records from URL in ssl_certificate and cache them
+	    ssl_stapling on;
+	    ssl_stapling_verify on;
+
+	    # Add headers to serve security related headers
+	    add_header X-Content-Type-Options nosniff;
+	    add_header X-Frame-Options "SAMEORIGIN";
+	    add_header X-XSS-Protection "1; mode=block";
+	    add_header X-Robots-Tag none;
+	    add_header X-Download-Options noopen;
+	    add_header X-Permitted-Cross-Domain-Policies none;
+
+	    location / {
+	        proxy_pass http://127.0.0.1:8008/;
+	        proxy_http_version 1.1;
+	        proxy_set_header Upgrade $http_upgrade;
+	        proxy_set_header Connection "upgrade";
+	        proxy_set_header Host $http_host;
+
+	        proxy_set_header X-Real-IP $remote_addr;
+	        proxy_set_header X-Forward-For $proxy_add_x_forwarded_for;
+	        proxy_set_header X-Forward-Proto http;
+	        proxy_set_header X-Nginx-Proxy true;
+
+	        proxy_redirect off;
+	    }
+	}
+	```
+
+1. Start serving the site by symlinking:
+
+	```
+	ln -s /etc/nginx/sites-available/matrix.tomesh.net /etc/nginx/sites-enabled/matrix.tomesh.net
+	```
+1. Reload nginx `service nginx reload`.
+
+1. Assuming Diffie-Hellman **.pem** and **crontab** for letsencrypt renewals are already configured, just run:
+
+	```
+	# /opt/letsencrypt/letsencrypt-auto certonly --agree-tos --renew-by-default --email hello@tomesh.net -a webroot --webroot-path=/usr/share/nginx/html -d matrix.tomesh.net
+	```
+
+### Update Synapse Version
+
+1. Enter the virtualenv from a root shell:
+
+	```
+	# cd ~/.synapse
+	# source ./bin/activate
+	```
+1. Stop the Synapse server with `synctl stop`.
+
+1. Update with the following command where `VERSION` can be a branch like `master` or `develop`, or a release tag like `v0.17.1`, or a commit hash:
+
+	```
+	# pip install --upgrade --process-dependency-links https://github.com/matrix-org/synapse/tarball/VERSION
+	```
+
+1. Start the Synapse server again with `synctl start`.
+
+## Set Up Vector Web Client
+
+The web client we host at **chat.tomesh.net** is running [Vector Web](https://github.com/vector-im/vector-web), and defaults to use our Matrix homeserver.
+
+### Build Vector Web From Source
+
+1. Use [nvm](https://github.com/creationix/nvm) to get these versions of **node.js** and **npm**:
+
+	```
+	$ node -v
+	v6.4.0
+	$ npm -v
+	3.10.3
```

Comment:
Another look at Vector's `readme` mentions this. Not sure if it's accurate from our experience:
> Install or update `node.js` so that your `npm` is at least at version `2.0.0`